astropy/io/misc/asdf/tags/helpers.py

```diff
@@ -7,7 +7,7 @@
 
 
 def skycoord_equal(sc1, sc2):
-    """SkyCoord equality useful for testing and ASDF serialization
+    """SkyCoord equality useful for testing
```

Comment:
Instead of duplicating code, might as well just call the function that you moved to `coordinates` here?